pkg/password/password_policies.go

```diff
@@ -0,0 +1,152 @@
+package password
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+	"unicode/utf8"
+)
+
+var _defaultSpecialCharacters = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~"
+
+// Validator describes the interface providing a password Validate method
+type Validator interface {
+	Validate(str string) error
+}
+
+// Policies represents a password validation rules
+type Policies struct {
+	minCharacters            int
+	minLowerCaseCharacters   int
+	minUpperCaseCharacters   int
+	minDigits                int
+	minSpecialCharacters     int
+	allowedSpecialCharacters string
+	digits                   *regexp.Regexp
+	specialCharacters        *regexp.Regexp
+}
+
+// NewPasswordPolicies returns a new NewPasswordPolicies instance
+func NewPasswordPolicies(minCharacters, minLowerCaseCharacters, minUpperCaseCharacters, minDigits, minSpecialCharacters int,
+	allowedSpecialCharacters string) (Validator, error) {
+
+	digits := regexp.MustCompile("[0-9]")
+	if len(allowedSpecialCharacters) > 0 {
+		_defaultSpecialCharacters = allowedSpecialCharacters
+	}
+	specialCharacters, err := regexp.Compile("[" + regexp.QuoteMeta(_defaultSpecialCharacters) + "]")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Please don't overwrite the default. This can lead to race conditions or misconfigurations. Do it other way around instead:
```suggestion
	if len(allowedSpecialCharacters) == 0 {
		allowedSpecialCharacters = _defaultSpecialCharacters
	}
	specialCharacters, err := regexp.Compile("[" + regexp.QuoteMeta(allowedSpecialCharacters) + "]")
	if err != nil {
		return nil, err
	}
```